<img src=".\images\Logo.png" width=150 align="left" /> <img src=".\images\Logo2.jpg" width=450 align="right" />


# <center><font color= #1e8449 > <b>CU18_Infraestructuras_eventos</font></center>

<font color='steelblue'><b>Citizenlab Data Science Methodology > II - Data Processing Domain </font>
***
> # <font color='steelblue'> <b>05.- Data Collection</font>

<font color='steelblue'>Data Collection is the process to obtain and generate (if required) necessary data to model the problem.</font>

### 16. Unión de todos los datos de distritos

* Unir todas los ficheros que contienen datos agregados de distritos







<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Settings" data-toc-modified-id="Settings-1"><font color="green">Settings</font></a></span></li><li><span><a href="#Data-Load" data-toc-modified-id="Data-Load-2"><font color="green">Data Load</font></a></span></li><li><span><a href="#ETL-Processes" data-toc-modified-id="ETL-Processes-3">ETL Processes</a></span><ul class="toc-item"><li><span><a href="#Import-data-from:-CSV,-Excel,-Tab,-JSON,-SQL,-and-Parquet-files" data-toc-modified-id="Import-data-from:-CSV,-Excel,-Tab,-JSON,-SQL,-and-Parquet-files-3.1">Import data from: CSV, Excel, Tab, JSON, SQL, and Parquet files</a></span></li></ul></li><li><span><a href="#Synthetic-Data-Generation" data-toc-modified-id="Synthetic-Data-Generation-4">Synthetic Data Generation</a></span></li><li><span><a href="#Fake-Data-Generation" data-toc-modified-id="Fake-Data-Generation-5">Fake Data Generation</a></span></li><li><span><a href="#Open-Data" data-toc-modified-id="Open-Data-6">Open Data</a></span></li><li><span><a href="#Data-Save" data-toc-modified-id="Data-Save-7"><font color="green">Data Save</font></a></span></li><li><span><a href="#Main-Conclusions" data-toc-modified-id="Main-Conclusions-8"><font color=" #2874a6 ">Main Conclusions</font></a></span></li><li><span><a href="#Main-Actions" data-toc-modified-id="Main-Actions-9"><font color=" #2874a6 ">Main Actions</font></a></span><ul class="toc-item"><li><ul class="toc-item"><li><span><a href="#Acciones-done" data-toc-modified-id="Acciones-done-9.0.1"><font color=" #2874a6 ">Acciones done</font></a></span></li><li><span><a href="#Acctions-to-perform" data-toc-modified-id="Acctions-to-perform-9.0.2"><font color=" #2874a6 ">Acctions to perform</font></a></span></li></ul></li></ul></li></ul></div>

## <font color='green'>Settings</font>

### Encoding

Con la siguiente expresión se evitan problemas con el encoding al ejecutar el notebook. Es posible que deba ser eliminada o adaptada a la máquina en la que se ejecute el código.

In [1]:
Sys.setlocale(category = "LC_ALL", locale = "es_ES.UTF-8")

[1] "es_ES.UTF-8/es_ES.UTF-8/es_ES.UTF-8/C/es_ES.UTF-8/C"

### Packages to use

- {tcltk} para selección interactiva de archivos locales
- {sf} para trabajar con georeferenciación
- {readr} para leer y escribir archivos csv
- {dplyr} para explorar datos


In [2]:
library(readr)
library(dplyr)



Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union




### Paths

In [3]:
iPath <- "Data/Input/"
oPath <- "Data/Output/"

## <font color='green'>Data Load</font>

If there are more than one input file, make as many sections as files to import.

Instrucciones
- Los ficheros de entrada del proceso están siempre en Data/Input/.   
- Si hay más de un fichero de entrada, se crean tantos objetos iFile_xx y file_data_xx como ficheros de entrada (xx número correlativo con dos dígitos, rellenar con ceros a la izquierda)

1. Infraestructuras

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 

Data load using the {tcltk} package. Ucomment the line if not using this option

In [4]:
# file_data_01 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>



In [5]:
iFile_01 <- "CU_18_05_05_01_infraestructuras_distrito.csv"
file_data_01 <- paste0(iPath, iFile_01)

if(file.exists(file_data_01)){
    cat("Se leerán datos del archivo: ", file_data_01)
} else{
    warning("Cuidado: el archivo no existe.")
}


Se leerán datos del archivo:  Data/Input/CU_18_05_05_01_infraestructuras_distrito.csv

#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [6]:
data_01 <- read_csv(file_data_01)


Rows: 239 Columns: 14
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (2): cmun, cdis
dbl (12): consultorios_de_salud, helisuperficies, centros_de_atencion_a_drog...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Estructura de  los datos:

In [7]:
glimpse(data_01)

Rows: 239
Columns: 14
$ cmun                                    <chr> "001", "002", "003", "004", "0…
$ cdis                                    <chr> "01", "01", "01", "01", "01", …
$ consultorios_de_salud                   <dbl> 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, …
$ helisuperficies                         <dbl> 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, …
$ centros_de_atencion_a_drogodependientes <dbl> 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, …
$ centros_de_salud                        <dbl> 0, 0, 0, 0, 2, 3, 2, 1, 2, 5, …
$ estaciones_de_cercanias                 <dbl> 0, 0, 0, 0, 1, 0, 1, 1, 0, 2, …
$ hospitales                              <dbl> 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, …
$ otros_centros_de_salud                  <dbl> 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, …
$ centros_de_salud_mental                 <dbl> 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, …
$ centros_de_especialidades               <dbl> 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, …
$ bocas_de_metro                          <dbl> 0, 0, 0, 0, 0, 0, 0, 0, 0, 9, …
$ intercambiadores

Muestra de datos:

2. Servicios

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 

Data load using the {tcltk} package. Ucomment the line if not using this option

In [8]:
# file_data_02 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>



In [9]:
iFile_02 <- "CU_18_05_09_01_distritos_adm.csv"
file_data_02 <- paste0(iPath, iFile_02)

if(file.exists(file_data_02)){
    cat("Se leerán datos del archivo: ", file_data_02)
} else{
    warning("Cuidado: el archivo no existe.")
}


Se leerán datos del archivo:  Data/Input/CU_18_05_09_01_distritos_adm.csv

#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [10]:
data_02 <- read_csv(file_data_02)


Rows: 246 Columns: 49
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (2): cmun, cdis
dbl (47): ayuntamientos_consejerias_ministerios_etc, sanidad_y_servicios_soc...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Estructura de  los datos:

In [11]:
glimpse(data_02)

Rows: 246
Columns: 49
$ cmun                                                      <chr> "001", "002"…
$ cdis                                                      <chr> "01", "01", …
$ ayuntamientos_consejerias_ministerios_etc                 <dbl> 1, 1, 1, 1, …
$ sanidad_y_servicios_sociales                              <dbl> 1, 8, 0, 10,…
$ actividades_administrativas                               <dbl> 0, 13, 0, 2,…
$ actividades_asociativas_y_hogares                         <dbl> 0, 2, 0, 2, …
$ actividades_profesionales                                 <dbl> 0, 11, 1, 3,…
$ administraciones_publicas                                 <dbl> 0, 2, 2, 2, …
$ alimentacion                                              <dbl> 0, 2, 0, 0, …
$ centros_educativos_no_universitarios_centros_publicos     <dbl> 0, 3, 0, 4, …
$ comercio_mayorista                                        <dbl> 0, 34, 0, 3,…
$ comercio_minorista                                        <dbl> 0, 22, 0, 10…
$ construccion    

Muestra de datos:

In [12]:
data_02 |> slice_head(n = 5)

cmun cdis ayuntamientos_consejerias_ministerios_etc
1 001  01   1                                        
2 002  01   1                                        
3 003  01   1                                        
4 004  01   1                                        
5 005  01   1                                        
  sanidad_y_servicios_sociales actividades_administrativas
1  1                            0                         
2  8                           13                         
3  0                            0                         
4 10                            2                         
5 42                           36                         
  actividades_asociativas_y_hogares actividades_profesionales
1  0                                 0                       
2  2                                11                       
3  0                                 1                       
4  2                                 3                       
5 14                                56                       
  administraciones_publicas alimentacion
1 0                         0           
2 2                         2           
3 2                         0           
4 2                         0           
5 5                         4           
  centros_educativos_no_universitarios_centros_publicos ...
1 0                                                     ...
2 3                                                     ...
3 0                                                     ...
4 4                                                     ...
5 8                                                     ...
  galerias_de_alimentacion grandes_superficies_especializadas hipermercados
1 0                        0                                  0            
2 0                        0                                  0            
3 0                        0                                  0            
4 0                        0                                  0            
5 3                        1                                  1            
  mercados_de_abastos agricultura_y_ganaderia
1 0                   0                      
2 0                   0                      
3 0                   0                      
4 0                   0                      
5 1                   0                      
  centros_educativos_no_universitarios_servicios_educativos seguridad_social
1 0                                                         0               
2 0                                                         0               
3 0                                                         0               
4 0                                                         0               
5 0                                                         0               
  centros_comerciales oficinas_de_empleo embajadas_y_consulados
1 0                   0                  0                     
2 0                   0                  0                     
3 0                   0                  0                     
4 0                   0                  0                     
5 0                   0                  0

3. Indicadores

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 

Data load using the {tcltk} package. Ucomment the line if not using this option

In [13]:
# file_data_03 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>



In [14]:
iFile_03 <- "CU_18_05_10_indicadores_distritos.csv"
file_data_03 <- paste0(iPath, iFile_03)

if(file.exists(file_data_03)){
    cat("Se leerán datos del archivo: ", file_data_03)
} else{
    warning("Cuidado: el archivo no existe.")
}


Se leerán datos del archivo:  Data/Input/CU_18_05_10_indicadores_distritos.csv

#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [15]:
data_03 <- read_csv(file_data_03)


Rows: 246 Columns: 22
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (2): CMUN, dist
dbl (20): nsec, t3_1, t1_1, t2_1, t2_2, t4_1, t4_2, t4_3, t5_1, t6_1, t7_1, ...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Estructura de  los datos:

In [16]:
glimpse(data_03)

Rows: 246
Columns: 22
$ CMUN            <chr> "001", "002", "003", "004", "005", "005", "005", "005"…
$ dist            <chr> "01", "01", "01", "01", "01", "02", "03", "04", "05", …
$ nsec            <dbl> 1, 2, 1, 4, 23, 36, 16, 18, 32, 67, 19, 37, 29, 35, 1,…
$ t3_1            <dbl> NA, 39.40747, 47.45970, 41.46662, 45.99877, 42.75131, …
$ t1_1            <dbl> 55, 4793, 248, 9946, 31006, 54049, 28117, 38778, 43620…
$ t2_1            <dbl> NA, 0.4777851, 0.3952000, 0.5015109, 0.5298303, 0.5115…
$ t2_2            <dbl> NA, 0.5222149, 0.6048000, 0.4984891, 0.4701697, 0.4884…
$ t4_1            <dbl> NA, 0.1696289, 0.1371000, 0.1810684, 0.1141978, 0.1468…
$ t4_2            <dbl> NA, 0.7218958, 0.6573000, 0.6420633, 0.6427127, 0.6644…
$ t4_3            <dbl> NA, 0.10847530, 0.20560000, 0.17684664, 0.24307467, 0.…
$ t5_1            <dbl> NA, 0.15562992, 0.11290000, 0.15474242, 0.19647849, 0.…
$ t6_1            <dbl> NA, 0.1959363, 0.1411000, 0.1912543, 0.2400069, 0.2147…
$ t7_1            

Muestra de datos:

In [17]:
data_03 |> slice_head(n = 5)

CMUN dist nsec t3_1     t1_1  t2_1      t2_2      t4_1      t4_2     
1 001  01    1         NA    55        NA        NA        NA        NA
2 002  01    2   39.40747  4793 0.4777851 0.5222149 0.1696289 0.7218958
3 003  01    1   47.45970   248 0.3952000 0.6048000 0.1371000 0.6573000
4 004  01    4   41.46662  9946 0.5015109 0.4984891 0.1810684 0.6420633
5 005  01   23   45.99877 31006 0.5298303 0.4701697 0.1141978 0.6427127
  t4_3      ... t7_1       t8_1       t9_1      t10_1     t11_1     t12_1    
1        NA ...         NA         NA        NA        NA        NA        NA
2 0.1084753 ... 0.41318314 0.40995322 0.4407990 0.1071522 0.6008132 0.6730932
3 0.2056000 ... 0.15420000 0.14490000 0.5280000 0.0932000 0.5000000 0.5514000
4 0.1768466 ... 0.15732451 0.15046840 0.2942034 0.1667587 0.4777910 0.5729139
5 0.2430747 ... 0.07438075 0.06545227 0.3850913 0.1412913 0.4565679 0.5316057
  X         Y        densidad_hab_km area_km2 
1 -3.635710 41.09315     2.514002    21.877471
2 -3.480171 40.52396   242.602224    19.756620
3 -3.849961 40.92033     9.647117    25.707163
4 -3.989259 40.23240   451.618447    22.023015
5 -3.364638 40.48268 17549.383203     1.766786

4. POIS

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 

Data load using the {tcltk} package. Ucomment the line if not using this option

In [18]:
# file_data_04 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>



In [19]:
iFile_04 <- "CU_18_05_13_01_ditrito_pois.csv"
file_data_04 <- paste0(iPath, iFile_04)

if(file.exists(file_data_04)){
    cat("Se leerán datos del archivo: ", file_data_04)
} else{
    warning("Cuidado: el archivo no existe.")
}


Se leerán datos del archivo:  Data/Input/CU_18_05_13_01_ditrito_pois.csv

#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [20]:
data_04 <- read_csv(file_data_04)


Rows: 224 Columns: 60
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (2): cmun, cdis
dbl (58): bakery, bar, butcher, cafe, car_dealership, clothes, convenience, ...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Estructura de  los datos:

In [21]:
glimpse(data_04)

Rows: 224
Columns: 60
$ cmun              <chr> "002", "003", "004", "005", "005", "005", "005", "00…
$ cdis              <chr> "01", "01", "01", "01", "02", "03", "04", "05", "01"…
$ bakery            <dbl> 2, 0, 0, 4, 5, 1, 2, 7, 4, 1, 1, 0, 2, 0, 0, 1, 0, 0…
$ bar               <dbl> 2, 0, 1, 22, 10, 2, 3, 11, 7, 1, 1, 2, 6, 0, 2, 17, …
$ butcher           <dbl> 1, 0, 0, 2, 2, 0, 1, 2, 0, 0, 1, 0, 0, 0, 0, 2, 0, 0…
$ cafe              <dbl> 5, 1, 0, 12, 6, 1, 3, 19, 12, 3, 2, 0, 6, 0, 1, 4, 0…
$ car_dealership    <dbl> 1, 0, 0, 1, 1, 3, 0, 3, 5, 1, 0, 0, 0, 0, 0, 0, 0, 0…
$ clothes           <dbl> 1, 0, 0, 13, 1, 0, 1, 28, 0, 0, 0, 0, 0, 0, 1, 1, 0,…
$ convenience       <dbl> 2, 0, 0, 4, 2, 2, 2, 13, 8, 1, 1, 2, 6, 0, 0, 1, 0, …
$ fast_food         <dbl> 1, 0, 0, 8, 1, 2, 5, 11, 10, 7, 4, 0, 6, 0, 2, 5, 0,…
$ food_court        <dbl> 2, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0…
$ furniture_shop    <dbl> 2, 0, 0, 3, 0, 0, 1, 8, 2, 1, 0, 0, 0, 0, 0, 0, 0, 0…
$ gift_shop       

Muestra de datos:

In [22]:
data_04 |> slice_head(n = 5)

cmun cdis bakery bar butcher cafe car_dealership clothes convenience
1 002  01   2       2  1        5   1               1      2          
2 003  01   0       0  0        1   0               0      0          
3 004  01   0       1  0        0   0               0      0          
4 005  01   4      22  2       12   1              13      4          
5 005  02   5      10  2        6   1               1      2          
  fast_food ... vending_parking chalet shelter motel camp_site general
1 1         ... 0               0      0       0     0         0      
2 0         ... 0               0      0       0     0         0      
3 0         ... 0               0      0       0     0         0      
4 8         ... 0               0      0       0     0         0      
5 1         ... 0               0      0       0     0         0      
  alpine_hut video_shop vending_machine car_sharing
1 0          0          0               0          
2 0          0          0               0          
3 0          0          0               0          
4 0          0          0               0          
5 0          0          0               0

5. AEMET

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 

Data load using the {tcltk} package. Ucomment the line if not using this option

In [23]:
# file_data_05 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>



In [24]:
iFile_05 <- "CU_18_05_15_distritos_meteo.csv"
file_data_05 <- paste0(iPath, iFile_05)

if(file.exists(file_data_05)){
    cat("Se leerán datos del archivo: ", file_data_05)
} else{
    warning("Cuidado: el archivo no existe.")
}


Se leerán datos del archivo:  Data/Input/CU_18_05_15_distritos_meteo.csv

#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [25]:
data_05 <- read_csv(file_data_05)


Rows: 246 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (2): CMUN, CDIS
dbl (4): tmed, prec, velmedia, presMax

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Estructura de  los datos:

In [26]:
glimpse(data_05)

Rows: 246
Columns: 6
$ CMUN     <chr> "001", "002", "003", "004", "005", "005", "005", "005", "005"…
$ CDIS     <chr> "01", "01", "01", "01", "01", "02", "03", "04", "05", "01", "…
$ tmed     <dbl> 11.47600, 15.31806, 12.00935, 15.13839, 15.27439, 15.31156, 1…
$ prec     <dbl> 3.6992270, 0.8713863, 2.0387942, 1.0548851, 0.8788330, 0.8511…
$ velmedia <dbl> 2.090249, 3.352072, 2.986153, 2.919731, 3.331738, 3.372923, 3…
$ presMax  <dbl> 904.2719, 946.6977, 880.7764, 931.9199, 945.8412, 946.8914, 9…


Muestra de datos:

In [27]:
data_05 |> slice_head(n = 5)

CMUN CDIS tmed     prec      velmedia presMax 
1 001  01   11.47600 3.6992270 2.090249 904.2719
2 002  01   15.31806 0.8713863 3.352072 946.6977
3 003  01   12.00935 2.0387942 2.986153 880.7764
4 004  01   15.13839 1.0548851 2.919731 931.9199
5 005  01   15.27439 0.8788330 3.331738 945.8412

## ETL Processes

### Import data from: CSV, Excel, Tab, JSON, SQL, and Parquet files

Se han importado en el apartado Data Load anterior:

* Infraestructuras (transporte y sanidad)
* Administración y servicios
* POIS (hostelería, comercio, turismo)
* Indicadores INE
* Datos meteorológicos


Incluir apartados si procede para: Extracción de datos (select, filter), Transformación de datos, (mutate, joins, ...). Si es necesario tratar datos perdidos, indicarlo también en NB 09.2

#### Data transformation

* Nombres de columna para unión consistentes

In [28]:
tdata_03 <- data_03 |>
    rename(cmun = CMUN,
            cdis = dist)
tdata_05 <- data_05 |>
    rename(cmun = CMUN,
            cdis = CDIS)



* Unión de todas las tablas


Si no aplica: Estos datos no requieren tareas de este tipo.

In [29]:
data <- data_01 |>
    full_join(data_02, by = c("cmun", "cdis")) |>
    full_join(tdata_03, by = c("cmun", "cdis")) |>
    full_join(data_04, by = c("cmun", "cdis")) |>
    full_join(tdata_05, by = c("cmun", "cdis"))

In [30]:
data |> glimpse()

Rows: 247
Columns: 143
$ cmun                                                      <chr> "001", "002"…
$ cdis                                                      <chr> "01", "01", …
$ consultorios_de_salud                                     <dbl> 1, 1, 1, 1, …
$ helisuperficies                                           <dbl> 1, 0, 1, 0, …
$ centros_de_atencion_a_drogodependientes                   <dbl> 0, 0, 0, 0, …
$ centros_de_salud                                          <dbl> 0, 0, 0, 0, …
$ estaciones_de_cercanias                                   <dbl> 0, 0, 0, 0, …
$ hospitales                                                <dbl> 0, 0, 0, 0, …
$ otros_centros_de_salud                                    <dbl> 0, 0, 0, 0, …
$ centros_de_salud_mental                                   <dbl> 0, 0, 0, 0, …
$ centros_de_especialidades                                 <dbl> 0, 0, 0, 0, …
$ bocas_de_metro                                            <dbl> 0, 0, 0, 0, …
$ intercambiadore

In [31]:
data |> slice_head(n = 5)

cmun cdis consultorios_de_salud helisuperficies
1 001  01   1                     1              
2 002  01   1                     0              
3 003  01   1                     1              
4 004  01   1                     0              
5 005  01   0                     0              
  centros_de_atencion_a_drogodependientes centros_de_salud
1 0                                       0               
2 0                                       0               
3 0                                       0               
4 0                                       0               
5 1                                       2               
  estaciones_de_cercanias hospitales otros_centros_de_salud
1 0                       0          0                     
2 0                       0          0                     
3 0                       0          0                     
4 0                       0          0                     
5 1                       1          1                     
  centros_de_salud_mental ... camp_site general alpine_hut video_shop
1 0                       ... NA        NA      NA         NA        
2 0                       ...  0         0       0          0        
3 0                       ...  0         0       0          0        
4 0                       ...  0         0       0          0        
5 0                       ...  0         0       0          0        
  vending_machine car_sharing tmed     prec      velmedia presMax 
1 NA              NA          11.47600 3.6992270 2.090249 904.2719
2  0               0          15.31806 0.8713863 3.352072 946.6977
3  0               0          12.00935 2.0387942 2.986153 880.7764
4  0               0          15.13839 1.0548851 2.919731 931.9199
5  0               0          15.27439 0.8788330 3.331738 945.8412

## Synthetic Data Generation

No aplica

## Fake Data Generation

No aplica

## Open Data

Los datos fueron descargados de datos abiertos (INE, CM, OSM, AEMET)

## <font color='green'>Data Save</font>

Este proceso, puede copiarse y repetirse en aquellas partes del notebbok que necesiten guardar datos.
Recuerde cambiar las cadenas añadida del fichero para diferenciarlas

<font color='tomato'> Identificamos los datos a guardar</font>

In [32]:
data_to_save <- data


<font color='tomato'>Estructura de nombre de archivos:</font>

* Código del caso de uso, por ejemplo "CU_04"
* Número del proceso que lo genera, por ejemplo "_05".
* Número de la tarea que lo genera, por ejemplo "_01"
* En caso de generarse varios ficheros en la misma tarea, llevarán _01 _02 ... después
* Nombre: identificativo de "properData", por ejemplo "_zonasgeo"
* Extensión del archivo

Ejemplo: "CU_04_05_01_01_zonasgeo.json, primer fichero que se genera en la tarea 01 del proceso 05 (Data Collection) para el caso de uso 04 (vacunas)

Importante mantener los guiones bajos antes de proceso, tarea, archivo y nombre

### Proceso 05

In [33]:
caso <- "CU_18"
proceso <- '_05'
tarea <- "_16"
archivo <- ""
proper <- "_distritos_variables"
extension <- ".csv"

<font color='tomato'><b> OPCION A:</b> Uso del paquete "tcltk" para mayor comodidad</font>

* Buscar carpeta, escribir nombre de archivo SIN extensión (se especifica en el código)
* Especificar sufijo2 si es necesario
* Cambiar datos por datos_xx si es necesario

In [34]:
# file_save <- paste0(caso, proceso, tarea, tcltk::tkgetSaveFile(), proper, extension) 
# path_out <- paste0(oPath, file_save)
# write_csv(data_to_save, path_out)

# cat('File saved as: ')
# path_out

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

- Los ficheros de salida del proceso van siempre a Data/Output/.  

In [35]:
file_save <- paste0(caso, proceso, tarea, archivo, proper, extension) 
path_out <- paste0(oPath, file_save)
write_csv(data_to_save, path_out)

cat('File saved as: ')
path_out

File saved as: 

[1] "Data/Output/CU_18_05_16_distritos_variables.csv"

#### Copia del fichero a Input

Si el archivo se va a usar en otros notebooks, copiar a la carpeta Input

In [36]:
path_in <- paste0(iPath, file_save)
file.copy(path_out, path_in, overwrite = TRUE)

[1] TRUE

## <font color=' #2874a6 '>Main Conclusions</font>

<font color='steelblue'>List and describe the general conclusions of the analysis carried out​.</font>

### Prerequisites

This working code needs the following conditions:

* For using the interactive selection of file, the {tcltk} package must be installed. It is not needed in production.
* The {readr} and {dplyr} packages must be installed. 
* The data paths `Data/Input` and `Data/Output` must exist (relative to the notebook path)

### Configuration Management

This notebook has been tested with the following versions of R and packages. It cannot be assured that later versions work in the same way:
* R 4.2.2
* tcltk 4.2.2
* readr 2.1.3
* dplyr 1.0.10

### Data structures

#### Objeto `data`

- Hay 248 filas con 143 columnas (ver tareas anteriores para detalle)


#### Observaciones generales sobre los datos

* Hay valores perdidos ya que algunos de los distritos no tienen datos en todos los archivos de datos

### Consideraciones para despliegue en piloto

* Ninguna

### Consideraciones para despliegue en producción

* Se deben crear los procesos ETL en producción necesarios para que los datos de entrada estén actualizados

## <font color=' #2874a6 '>Main Actions</font>

#### <font color=' #2874a6 '>Acciones done</font>

<font color='steelblue'>Indicate the actions that have been carried out in this process</font>

- Se han unido las tablas con información por distritos

#### <font color=' #2874a6 '>Acctions to perform</font>

<font color='steelblue'>Indicate the actions that must be carried out in subsequent processes</font>

- Se debe decidir si imputar valores perdidos u omitirlos en cada modelo

## <font color='RED'>CODE TO DEPLOY (PILOT)</font>

A continuación se incluirá el código que deba ser llevado a despliegue para producción, dado que se entiende efectúa operaciones necesarias sobre los datos en la ejecución del prototipo

<font color=' red '><b>Description</b></font>

- No hay nada que desplegar en el piloto, ya que estos datos son estáticos o en todo caso cambian con muy poca frecuencia, altamente improbable durante el proyecto.


<font color=' red '><b>CODE</b></font>

In [37]:
# incluir código